pkg/archive/archive_test.go

```diff
@@ -287,8 +287,15 @@ func TestUntarPathWithInvalidDest(t *testing.T) {
 	// Create a src file
 	srcFile := filepath.Join(tempFolder, "src")
 	tarFile := filepath.Join(tempFolder, "src.tar")
-	os.Create(srcFile)
-	os.Create(invalidDestFolder) // being a file (not dir) should cause an error
+	f, err := os.Create(srcFile)
```

Comment:
Only closing if `Check` is true? Why?

Reply:
If there's an error, it returns nil, so no need to close.

Reply:
Oh right, these are `Check`'s which will continue on.

Reply:
Ah! Yes, indeed. I try to move more towards `assert.Check()` instead of `assert.Assert()`. We have too many tests where we fail too early (and could continue with at least some of the steps, which could give more details and help debugging failures / flakiness)